api/src/main/java/org/openmrs/module/patientflags/api/FlagService.java

```diff
@@ -139,6 +139,26 @@
 	@Transactional(readOnly = true)
 	@Authorized(value = { "Manage Flags", "Test Flags" }, requireAll = false)
 	public Flag getFlag(Integer flagId);
+
+	/**
+	 * Retrieve a specific Flag based on its uuid
+	 *
+	 * @param uuid
+	 * @return the flag which matches the specific uuid
+	 */
+	@Transactional(readOnly = true)
+	@Authorized(value = { "Manage Flags", "Test Flags" }, requireAll = false)
+	public Flag getFlagByUuid(String uuid);
+
+	/**
+	 * Retrieve a specific Flag based on its name
+	 *
+	 * @param name
+	 * @return the flag which matches the specific name
+	 */
+	@Transactional(readOnly = true)
+	@Authorized(value = { "Manage Flags", "Test Flags" }, requireAll = false)
+	public Flag getFlagByName(String name);

```

Comment:
Did you make sure there's an db index on name?

Reply:
@rkorytkowski Will add unique indexes to all the names - for flag, priority and tags